backend/pkg/api/handle_schema_registry_integration_test.go

```diff
@@ -0,0 +1,126 @@
+		assert.NotEmpty(validationResponse.ParsingError, "parsing error should be set")
+	})
+
+	t.Run("malformed schema (avro)", func(t *testing.T) {
+		ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+		defer cancel()
+
+		// Schema is missing the name property
+		schemaStr := "{\r\n  \"type\": \"record\",\r\n  \"namespace\": \"com.shop.v1.avro\",\r\n  \"doc\": \"Address is a customer's address that can be selected for deliveries or invoices\",\r\n  \"fields\": [\r\n    {\r\n      \"name\": \"version\",\r\n      \"type\": \"int\"\r\n    },\r\n    {\r\n      \"name\": \"id\",\r\n      \"type\": \"string\"\r\n    },\r\n    {\r\n      \"name\": \"customer\",\r\n      \"type\": {\r\n        \"name\": \"AddressCustomer\",\r\n        \"type\": \"record\",\r\n        \"fields\": [\r\n          {\r\n            \"name\": \"id\",\r\n            \"type\": \"string\"\r\n          },\r\n          {\r\n            \"name\": \"type\",\r\n            \"type\": \"string\"\r\n          }\r\n        ]\r\n      }\r\n    },\r\n    {\r\n      \"name\": \"type\",\r\n      \"type\": {\r\n        \"name\": \"AddressType\",\r\n        \"type\": \"enum\",\r\n        \"symbols\": [\"INVOICE\", \"DELIVERY\"]\r\n      }\r\n    },\r\n    {\r\n      \"name\": \"firstName\",\r\n      \"type\": \"string\"\r\n    },\r\n    {\r\n      \"name\": \"lastName\",\r\n      \"type\": \"string\"\r\n    },\r\n    {\r\n      \"name\": \"state\",\r\n      \"type\": \"string\"\r\n    },\r\n    {\r\n      \"name\": \"street\",\r\n      \"type\": \"string\"\r\n    },\r\n    {\r\n      \"name\": \"houseNumber\",\r\n      \"type\": \"string\"\r\n    },\r\n    {\r\n      \"name\": \"city\",\r\n      \"type\": \"string\"\r\n    },\r\n    {\r\n      \"name\": \"zip\",\r\n      \"type\": \"string\"\r\n    },\r\n    {\r\n      \"name\": \"latitude\",\r\n      \"type\": \"double\"\r\n    },\r\n    {\r\n      \"name\": \"longitude\",\r\n      \"type\": \"double\"\r\n    },\r\n    {\r\n      \"name\": \"phone\",\r\n      \"type\": \"string\"\r\n    },\r\n    {\r\n      \"name\": \"additionalAddressInfo\",\r\n      \"type\": \"string\"\r\n    },\r\n    {\r\n      \"name\": \"createdAt\",\r\n      \"type\": {\"type\": \"int\", \"logicalType\": \"date\"}\r\n    },\r\n    {\r\n      \"name\": \"revision\",\r\n      \"type\": \"int\"\r\n    }\r\n  ]\r\n}"
+		req := struct {
+			Schema string `json:"schema"`
+			Type   string `json:"schemaType"`
+		}{
+			Schema: schemaStr,
+			Type:   schema.TypeAvro.String(),
+		}
+		res, body := s.apiRequest(ctx, http.MethodPost, "/api/schema-registry/subjects/new-schema/versions/latest/validate", req)
+		require.Equal(200, res.StatusCode)
+
+		validationResponse := console.SchemaRegistrySchemaValidation{}
+		err := json.Unmarshal(body, &validationResponse)
+		require.NoError(err)
+
+		assert.False(validationResponse.IsValid)
+		assert.NotEmpty(validationResponse.ParsingError, "parsing error should be set")
+	})
+}
```

Comment:
We should perhaps add a follow up task to expand these integration tests to cover more of the schema registry APIs?